lib/game.rb

```diff
@@ -29,46 +31,48 @@ def set_options
       @human_player.marker = @marker
     end
 
+    def generate_markers
+      markers = []
+
+      @players.each do |player|
+        markers << player.marker
+      end
+
+      markers
+    end
+
     def run_game_loop
-      until is_game_over?
+      markers = generate_markers
+
+      until TicTacToe::is_game_over?(@board, markers)
         if is_computer_the_current_player?
           make_computer_move
         else
-          @user_interface.display_board(@board.grid)
-          move = @user_interface.select_move
-          begin
-            make_human_move(move)
-          rescue Exception => message
-            @user_interface.display_error(message)
-          end
+          make_human_move  
         end
       end
 
       @user_interface.display_board(@board.grid)
-      outcome = declare_outcome 
+      outcome = declare_outcome(@board, markers)
       @user_interface.display_outcome(outcome)
     end
 
-    def declare_outcome
-      if is_game_over?
-        if draw?
-          return DRAW
-        elsif win?
-          declare_winner
-        end
-      end
-    end
-
     private
-    
-    def is_game_over?
-      win? || draw?
-    end
 
     def is_computer_the_current_player?
       @current_player == @computer_player
     end
 
+    def make_human_move
```

Comment:
The game shouldn't know how a player moves since it isn't its responsibility to know how - only delegating to the players.